astropy/stats/circstats.py

```diff
@@ -0,0 +1,448 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+This module contains simple functions for dealing with circular statistics, for
+instance, mean, variance, standard deviation, correlation coefficient, and so
+on. This module also cover tests of uniformity, e.g., the Rayleigh and V tests.
+The Maximum Likelihood Estimator for the Von Mises distribution along with the
+Cramer-Rao Lower Bounds are also implemented. Almost all of the implementations
+are based on reference [1], which is also the basis for the R package
+'CircStats' [2].
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+from astropy.units import Quantity
+from astropy import units as u
+
+__all__ = ['circmean', 'circvar', 'circmoment', 'circcorrcoef', 'rayleightest',
+           'vtest', 'vonmisesmle']
+__doctest_requires__ = {'vtest': ['scipy.stats']}
+
+
+def _components(data, w=None, p=1, phi=0.0, axis=None):
+    # Utility function for computing the generalized rectangular components
+    # of the circular data.
+    if w is None:
```

Comment:
One could be slightly more general here, allowing the weights to be broadcasted:
```
if w is None:
    w = np.ones((1,))
try:
    w = broadcast_to(w, data.shape)
except ValueError:
    raise ValueError('Weights and data have inconsistent shape.')
```
(Note that you need to add `from astropy.utils.compat.numpy import broadcast_to` since it is only in recent numpy versions.)